### ISSUE TITLE
Adding Google Data Saver Proxy to Brave browser on Android

### ISSUE THREAD

**Issue by i-rme** [brave/browser-android-tabs#239]
Google Data Saver Proxy is a feature that can be enabled to save up to 50% of the data usage on mobile when browsing the web.

This feature has already being implemented on other platforms like Mozilla Firefox on an extension:

- https://addons.mozilla.org/es/firefox/addon/google_datasaver_for_firefox/

Third parties have implemented this even on Chrome on desktop in this extension:

- https://chrome.google.com/webstore/detail/data-compression-proxy/ajfiodhbiellfpcjjedhmmmpeeaebmep

There is even an official implementation on Chrome here:

- https://chrome.google.com/webstore/detail/data-saver/pfmgfdlgomnbgkofeojodiodmgpgmkac

Also you can find more information about the implementation here:

- https://superuser.com/questions/945924/how-do-i-use-the-google-data-compression-proxy-on-firefox/959419#959419

- https://developer.chrome.com/multidevice/data-compression

It seems that is easy to implement this feature as an opt-in setting so the user can enable it if he wants.

This feature would improve Brave browser because it would be the first android browser to feature Adblock and Data Compression at the same time.

Thanks.

**Comment by i-rme** [comment]
Implementing this should be easy because it is a simple HTTP proxy + adding a request parameter in each request.

**Comment by theel0ja** [comment]
I don't think that this is useful feature, since it supports only HTTP and [over 70% of requests](https://twitter.com/ow/status/993989804021792768) are HTTPS.

**Comment by NumDeP** [comment]
One thing many may not understand is that this process usually requires servers owned primarily by the host that supports the feature. So Google Data Saver - data is shrink-ed at Google's servers and Opera's Data Saver is shrink-ed at Opera's own servers (if it has it's own server park/farm that is).

My first point being, Brave will have to deploy their own server farm to shrink data for its users that use its browser and not everybody will utilise the feature because it is arguably inefficient.

Second point being, I personally don't think it is efficient. The only people I could see it helping are those that are data conscious, meaning those that are worried if they're going to go over their monthly data allowance **only** on their mobiles. If anything, it is more resource intensive and therefore takes more time for content to appear and for Brave to block/upgrade data because of the shrinking process. 

Overall, I don't think the Brave team will be comfortable with using third-party servers to handle their users' data for privacy reason and if they were to build their own server parks it will have to be in several countries owned by Brave completely (the parks not the countries :). In addition to being very costly, it will be too (to keep it short) against its current principle - Fast Ad&Track Blocker with a Great Privacy Initiative.

I think for several years to come, Brave will most likely settle for the Brotli compression (and any enhancements that come to that side of things) which is practically part of Chromium, I think.

It would be terribly embarrassing if I just got a thumbs down from a moderator.

**Comment by dirtbag** [comment]
https://venturebeat.com/2019/03/12/chromes-lite-pages-speed-up-https-webpages-on-slow-connections/
looks like the "they dont do https" argument is now irrelevant.. 

-db
